### PR TITLE
New RFC: Response Filters

### PR DESCRIPTION
**Pull request by eldadfux**

### CODE REVIEW

**Review by Glide7 — APPROVED** [review]
Looks good to me!

**Comment by christyjacob4** [comment]
@eldadfux  Could you attach some example responses **before and after** this change? Just trying to get a better idea of what the RFC proposes

File: 002-response-filters.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+### Create a Filter Interface
+
+Create a new Appwrite\Utopia\Response\Filter class (src/Appwrite/Utopia/Response/Filter.php). The interface should decalre an implemntation for the `parse(array $content): array` method. Each new filter will implement the interface and use the parse method to manipulate incoming data. Create our first filter (Appwrite\Utopia\Response\Filters\V06) to accept current version (v0.7) data and convert it to v0.6 data. For that we'll need to run both instances of Appwrite to manually convert data structures.
+
+### Run the Filter
+
+Add a piece of code in the response output stage, that will check if a filter is set and if it is will execute his `parse` method and alter the response output before it is returned to the client. A good location for this method to run might be: https://github.com/appwrite/appwrite/blob/0.7.x/src/Appwrite/Utopia/Response.php#L296
```
Comment (TorstenDittmann) [inline 002-response-filters.md]: `that will check if a filter is set and will`

small typo

**Review by TorstenDittmann — APPROVED** [review]
lgtm!

One question: Will this new header be set on default with the SDKs then?

File: 002-response-filters.md (hunk 1 above)
Comment (eldadfux) [reply 002-response-filters.md]: Fixed, thanks!

**Comment by eldadfux** [comment]
> lgtm!
> 
> One question: Will this new header be set on default with the SDKs then?

Might be a good idea. Is your idea to couple an SDK version with an expected API format?

For example: JS-SDK v1.0 will attach the header: `X-Appwrite-Response-Format: 0.7.0`, and future JS-SDK v2.0 will attach the header: `X-Appwrite-Response-Format: 0.8.0`?

**Comment by eldadfux** [comment]
> @eldadfux Could you attach some example responses **before and after** this change? Just trying to get a better idea of what the RFC proposes

Done ✅

**Pull request merged**